config/initializers/rack_attack.rb

```diff
@@ -1,23 +1,40 @@
 ##
 # See https://github.com/kickstarter/rack-attack/blob/master/docs/example_configuration.md
 # for more configuration options
-class Rack::Attack
-  ### Throttle Spammy Clients ###
 
-  # If any single client IP is making tons of requests, then they're
-  # probably malicious or a poorly-configured scraper. Either way, they
-  # don't deserve to hog all of the app server's CPU. Cut them off!
-  #
-  # Note: If you're serving assets through rack, those requests may be
-  # counted by rack-attack and this throttle may be activated too
-  # quickly. If so, enable the condition to exclude them from tracking.
+### Throttle Spammy Clients ###
 
+# If any single client IP is making tons of requests, then they're
+# probably malicious or a poorly-configured scraper. Either way, they
+# don't deserve to hog all of the app server's CPU. Cut them off!
+#
+# Note: If you're serving assets through rack, those requests may be
+# counted by rack-attack and this throttle may be activated too
+# quickly. If so, enable the condition to exclude them from tracking.
+
+if Settings.THROTTLE_TRAFFIC
   # Throttle all requests by IP (60rpm)
-  #
-  # Key: "rack::attack:#{Time.now.to_i/:period}:req/ip:#{req.ip}"
-  if Settings.THROTTLE_TRAFFIC
-    throttle('req/ip', limit: 300, period: 5.minutes) do |req|
-      req.ip unless req.path.start_with?('/assets')
-    end
+  Rack::Attack.throttle('req/ip', limit: 300, period: 5.minutes) do |req|
+    req.ip unless req.path.start_with?('/assets')
+  end
+
+  # Throttle article searching more aggressively to protect EDS API (30rpm)
+  Rack::Attack.throttle('articles/ip', limit: 150, period: 5.minutes) do |req|
```

Comment:
Just a thought: if we excluded Stanford IPs `171.64.0.0/14` from this throttling, maybe we could get away with being more aggressive -- something like 60/5min or 12rpm?

Reply:
Updated to use 60/5min